python/lsst/ts/tunablelaser/mock_server.py

```diff
@@ -1,45 +1,51 @@
 import logging
-import queue
-import serial
 import inspect
+import asyncio
 
+from lsst.ts import tcpip
 
-class MockSerial(serial.Serial):
-    def __init__(
-        self,
-        port,
-        baudrate=19200,
-        bytesize=serial.EIGHTBITS,
-        parity=serial.PARITY_NONE,
-        stopbits=serial.STOPBITS_ONE,
-        timeout=3,
-        xonxoff=False,
-        rtscts=False,
-        write_timeout=None,
-        dsrdtr=False,
-        inter_byte_timeout=None,
-        exclusive=None,
-    ):
-        super().__init__(port=port)
-        self.log = logging.getLogger(__name__)
 
+class MockServer(tcpip.OneClientServer):
+    """Simulates the TunableLaser."""
+
+    def __init__(self) -> None:
         self.device = MockNT900()
-        self.message_queue = queue.Queue()
+        self.log = logging.getLogger(__name__)
+        self.read_loop_task = asyncio.Future()
+        self.lock = asyncio.Lock()
```

Comment:
What are you doing with this lock? It appears that you fully process each command before reading another. If so, I suspect you don't need the lock.